src/commands/search.rs

```diff
@@ -32,16 +34,30 @@ fn run_search(directory: &Path, query: &str) -> tantivy::Result<()> {
     let query = query_parser.parse_query(query)?;
     let searcher = index.reader()?.searcher();
     let weight = query.weight(&searcher, false)?;
+
+    let stdout = io::stdout();
+    let mut handle = io::BufWriter::new(stdout);
+
     for segment_reader in searcher.segment_readers() {
         let mut scorer = weight.scorer(segment_reader, 1.0)?;
         let store_reader = segment_reader.get_store_reader();
         while scorer.doc() != TERMINATED {
             let doc_id = scorer.doc();
             let doc = store_reader.get(doc_id)?;
             let named_doc = schema.to_named_doc(&doc);
-            println!("{}", serde_json::to_string(&named_doc).unwrap());
+            if let Err(e) = writeln!(
+                handle,
+                "{}",
+                serde_json::to_string(&named_doc).unwrap()
+            ) {
+                if e.kind() != ErrorKind::BrokenPipe {
+                    eprintln!("{}", e.to_string());
+                    process::exit(1)
+                }
+            }
             scorer.advance();
         }
```

Comment:
Please flush handle. `Drop` does flush as well but it does not emit any error if there is an error.